### ISSUE TITLE
Assess existence of experimental models

### ISSUE THREAD

**Issue by bschilder** [neurogenomics/RareDiseasePrioritisation#33]
Assess whether there is an existing experimental model for each candidate therapeutics target.

We can check this by seeing if there is an MPO or UPHENO annotation for the same phenotype.

**Comment by bschilder** [comment]
Found a treasure trove of experimental models for diseases (and perhaps specific phenotypes) on Monarch:
https://data.monarchinitiative.org/latest/tsv/model_associations/

These file don't include gene-level info (which we would want if we have a particular gene therapy target in mind), but I'm checking to see if there's a way I can extract that from the larger Monarch knowledge graph:
https://data.monarchinitiative.org/monarch-kg/latest/

**Comment by NathanSkene** [comment]
What's argument against just using Mammalian Phenotype Ontology overlap?


Also, here's some of the messages we sent relating to this previously:

Here's one of the gene's that a mouse model for respiratory failure: http://www.informatics.jax.org/reference/J:120296

Here's the list of mammalian phenotype ontology genes (for respiratory failure): http://www.informatics.jax.org/mp/annotations/MP:0001953 (edited)

Gene therapy for ABCA3 in respiratory failure is already being looked into: https://www.ncbi.nlm.nih.gov/pmc/articles/PMC8798122/

**Comment by bschilder** [comment]
> What's argument against just using Mammalian Phenotype Ontology overlap?

Several reasons:
- Monarch includes MPO, as well as other model organism databases beyond just mouse.
- We need to map MPO to HPO terms. UPHENO provides this, which is also integrated in Monarch.

**Comment by NathanSkene** [comment]
Sounds good!

Sent from Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by NathanSkene** [comment]
Great, didn’t think about looking at zebrafish models etc as well!

Can you explain the x-axis?

From: Brian M. Schilder ***@***.***>
Date: Friday, 1 December 2023 at 11:40
To: neurogenomics/RareDiseasePrioritisation ***@***.***>
Cc: Skene, Nathan G ***@***.***>, Comment ***@***.***>
Subject: Re: [neurogenomics/RareDiseasePrioritisation] Assess existence of experimental models (Issue #33)
This email from ***@***.*** originates from outside Imperial. Do not click on links and attachments unless you recognise the sender. If you trust the sender, add them to your safe senders list<https://spam.ic.ac.uk/SpamConsole/Senders.aspx> to disable email stamping for this address.



Preliminary summary plot showing the proportion of orthologous genes overlapping between HPO and non-human ontology databases, across many phenotypes:

crossspecies-ontologies.png (view on web)<https://github.com/neurogenomics/RareDiseasePrioritisation/assets/34280215/dc62c8e0-17d1-48e3-99a7-667fd128e4ea>

Will include this in the final report as well as showing how we can use this to prioritise gene/phenotype-specific therapeutic targets.

**Comment by bschilder** [comment]
> Great, didn’t think about looking at zebrafish models etc as well! Can you explain the x-axis?

Sure! 
- `n_genes_intersect`: for a given phenotype that has a match between a pair of species, count the number of orthologous genes shared between the gene-phentypes annotations of each species.
- `n_genes_hpo`: the total number of unique human genes annotated for a given HPO phenotype.

Dividing one over the other thus gives you the proportion of HPO gene annotations recapitulated in the equivalent phenotype of another species. 

This proportion will be influenced by both evolutionary distance and how well studied each species is (notice the difference between mouse and rate, despite the fact that they're equally related to humans).

**Comment by bschilder** [comment]
Here's some gene therapy target phenotypes identified by our previous analyses. The exact phenotypes will likely change once we add chatGPT annotations to our filtering strategy with the round of enrichment results. But for now these can serve as an example.

 with the heatmap colored by the "equivalence score", which is essentially UPHENO's way of quantifying how well a phenotype matches up across species (on a scale from 0-1). Data comes from [here](https://github.com/obophenotype/upheno/tree/master/mappings).
> Currently the fuzzy equivalence score is the Jaccard similarity:

Not sure exactly on what basis they computed Jaccard similarity, but I'll look into this some more.



[upheno_top_targets_heatmap.pdf](https://github.com/neurogenomics/RareDiseasePrioritisation/files/13529548/upheno_top_targets_heatmap.pdf)

Looks like UPHENO has been thinking about adding fly ontology mappings as well, though there hasn't been any activity on this since 2016 it seems. Just pinged them to get an update:
- https://github.com/obophenotype/upheno/issues/207

**Comment by bschilder** [comment]
Met with @matentzn who was extremely helpful in explaining the matching procedure to me, and pointing me to some additional resources.


For mapping MONDO IDs in the Monarch [model's file](https://data.monarchinitiative.org/latest/tsv/model_associations/), I'm switching to using [this file](https://github.com/monarch-initiative/mondo/blob/master/src/ontology/mappings/mondo.sssom.tsv) as it avoid issues observed here:
- https://github.com/monarch-initiative/mondo/issues/6873

With these changes, `HPOExplorer` can now map >90% of MONDO ids listed in the model file to OMIM IDs:
```
library(HPOExplorer)
>  model <- get_monarch("disease_to_model")
 [100%] Downloaded 883280 bytes...
>   model$db <- stringr::str_split(model$subject,":", simplify = TRUE)[,1]
>   model <- map_mondo(dat = model,
+                      input_col="object",
+                      output_col="OMIM_ID",
+                      to=c("OMIM","Orphanet"))
 [100%] Downloaded 1082741 bytes...
476 / 5,154 (9.24%) OMIM_ID missing.
```

The only issue is, as far as I can tell MONDO doesn't seem to contain any mappings between MONDO IDs and DECIPHER IDs. DECIPHER IDs only make up a small fraction of the HPO annotations, but would be nice to have a complete mapping nonetheless:

![Screenshot 2023-12-08 at 23 12 22](https://github.com/neurogenomics/RareDiseasePrioritisation/assets/34280215/ae5e0036-d5d2-4c7a-8d48-33fb7e57c127)

**Comment by bschilder** [comment]
To summarise, the phenotype matching procedure is meant to captured semantic similarity using a semi-heuristic model (a combination of explicit rules and data-driven). Data inputs come from a variety of sources. Ultimately, they linking together concepts (species, diseases, phenotypes, genes, pathways, etc.) in a knowledge graph derived from a mix of NLP queries to the published literature and other database.

@matentzn this is probably a poor attempt to explain this properly, but if there's a paper or docs page you could point me to that would be quite helpful! Thanks!

**Comment by matentzn** [comment]
> DECIPHER

We have this for DECIPHER: https://github.com/monarch-initiative/mondo/blob/master/src/ontology/mappings/mondo_hasdbxref_decipher.sssom.tsv

Which will do the job for you!

> To summarise, the phenotype matching procedure is meant to captured semantic similarity using a semi-heuristic model (a combination of explicit rules and data-driven). Data inputs come from a variety of sources. Ultimately, they linking together concepts (species, diseases, phenotypes, genes, pathways, etc.) in a knowledge graph derived from a mix of NLP queries to the published literature and other database.

Its simpler than that.

1. We generate phenotypic profiles from ontologies, using jaccard similarity  usually over the hierarchical relations in the ontology and information content for the reranking 
2. Cool Paper: https://www.osti.gov/biblio/1625303 with background
3. The current "bestmatches" include a mix of logical and simple lexical matches and are hugely out of date (I would not use them in production, but they are probably "not wrong"

I requested an FBcv profile for you here: https://github.com/monarch-initiative/monarch-semantic-similarity-profiles/issues/16

So you can take a look how it looks like.

**Comment by bschilder** [comment]
> > DECIPHER
> 
> We have this for DECIPHER: https://github.com/monarch-initiative/mondo/blob/master/src/ontology/mappings/mondo_hasdbxref_decipher.sssom.tsv
> 
> Which will do the job for you!

Ah, amazing! I had totally missed that bc i was using this file, which I assumed included all the other ones:
https://github.com/monarch-initiative/mondo/blob/master/src/ontology/mappings/mondo.sssom.tsv

I've implemented many of these functions within a new package for accessing/processing knowledge graphs in general (HPOExplorer was getting to bloated):
https://github.com/neurogenomics/KGExplorer/blob/29eccbbd33fd18d9ce85b0ae72b47d485d97faee/R/map_upheno_data_i.R

I was also just alerted to the `monarchr` package, which may extract much of the info i need more efficiently than I am now (which relies mostly on TSV downloads).
- https://github.com/monarch-initiative/monarchr/issues/7

I've also begun exploring some of the graph query resources/tools you alerted to me on our call:
- https://github.com/neurogenomics/KGExplorer/issues/1


> > To summarise, the phenotype matching procedure is meant to captured semantic similarity using a semi-heuristic model (a combination of explicit rules and data-driven). Data inputs come from a variety of sources. Ultimately, they linking together concepts (species, diseases, phenotypes, genes, pathways, etc.) in a knowledge graph derived from a mix of NLP queries to the published literature and other database.
> 
> Its simpler than that.
> 
> 1. We generate phenotypic profiles from ontologies, using jaccard similarity  usually over the hierarchical relations in the ontology and information content for the reranking
> 2. Cool Paper: https://www.osti.gov/biblio/1625303 with background
> 3. The current "bestmatches" include a mix of logical and simple lexical matches and are hugely out of date (I would not use them in production, but they are probably "not wrong"

Ahhh, this makes so much more sense now! Thanks for explaining that in more detail, and for the paper (super interesting work!). Along those lines, I've found the [`rphenoscape`](https://github.com/phenoscape/rphenoscape) package useful for computing cross-ontology similarity matrices on the go.


> I requested an FBcv profile for you here: [monarch-initiative/monarch-semantic-similarity-profiles#16](https://github.com/monarch-initiative/monarch-semantic-similarity-profiles/issues/16)
> 
> So you can take a look how it looks like.

Thank you so much! I really appreciate this, and all your other help.
